validation/genesis-predeploy_test.go

```diff
@@ -0,0 +1,121 @@
+package validation
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+// Define a struct to represent the structure of the JSON data
+type DeployedBytecode struct {
+	Object              string                          `json:"object"`
+	ImmutableReferences map[string][]ImmutableReference `json:"immutableReferences"`
+}
+
+type ImmutableReference struct {
+	Start  int `json:"start"`
+	Length int `json:"length"`
+}
+
+type ContractData struct {
+	DeployedBytecode DeployedBytecode `json:"deployedBytecode"`
+}
+
+// Invoke this with go test -timeout 0 ./validation -run=TestGenesisPredeploys -v
+// REQUIREMENTS:
+// yarn, so we can prepare https://codeload.github.com/Saw-mon-and-Natalie/clones-with-immutable-args/tar.gz/105efee1b9127ed7f6fedf139e1fc796ce8791f2
+func TestGenesisPredeploys(t *testing.T) {
+
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("No caller information")
+	}
+	// Get the directory of the current file
+	dir := filepath.Dir(filename)
+
+	monorepoDir := path.Join(dir, "../../optimism")
+	contractsDir := path.Join(monorepoDir, "packages/contracts-bedrock")
+
+	chainId := uint64(34443) // Mode mainnet
+
+	monorepoCommit := "d80c145e0acf23a49c6a6588524f57e32e33b91c"
```

Comment:
This commit was provided by the security team as the commit used by conduit to generate the genesis for Mode.